code/game/machinery/computer/shuttle.dm

```diff
@@ -31,23 +34,25 @@
 			return 0
 		switch(choice)
 			if("Authorize")
-				src.authorized -= W:registered_name
-				src.authorized += W:registered_name
-				if (src.auth_need - src.authorized.len > 0)
-					message_admins("[key_name(user.client)](<A HREF='?_src_=holder;adminmoreinfo=\ref[user]'>?</A>) has authorized early shuttle launch in ([x],[y],[z] - <A HREF='?_src_=holder;adminplayerobservecoodjump=1;X=[x];Y=[y];Z=[z]'>JMP</a>)",0,1)
-					log_game("[user.ckey]([user]) has authorized early shuttle launch in ([x],[y],[z])")
-					minor_announce("[src.auth_need - src.authorized.len] more authorization(s) needed until shuttle is launched early",null,1)
+				if (emergency_shuttle.location == DOCKED)

```

Comment:
check after the authorized button is pressed to see if the shuttle is at dock then executes based on that.